src/persist/src/indexed/cache.rs

```diff
@@ -555,27 +562,29 @@ impl<D> BlobCacheCore<D> {
     /// size limit.
     fn add(&mut self, key: String, size: usize, data: Arc<D>) {
         let new_size = self.size + size;
-
         if new_size > self.limit {
             return;
         }
 
-        let prev = self.dataz.insert(key, data);
+        let prev = self.dataz.insert(key, (data, size));
         if prev.is_none() {
             self.size = new_size;
         }
+
+        debug_assert!(self.limit >= self.size);
     }
 
-    fn remove(&mut self, key: &str, size: usize) {
+    fn remove(&mut self, key: &str) {
         let prev = self.dataz.remove(key);
-
-        if prev.is_some() {
+        if let Some((_, size)) = prev {
```

Comment:
`debug_assert!(self.size > size)` before the subtraction?

Reply:
i believe that should be `self.size >= size` right?

Reply:
done